File: packages/shared-ts/src/env.ts
/** Env var access with explicit failure on missing required values. */

export function required(name: string): string {
  const v = process.env[name];
  if (!v) throw new Error(`missing required env var: ${name}`);
  return v;
}

export function optional(name: string, fallback = ""): string {
  return process.env[name] ?? fallback;
}

export function optionalNumber(name: string, fallback: number): number {
  const v = process.env[name];
  if (v === undefined || v === "") return fallback;
  const n = Number(v);
  if (!Number.isFinite(n)) throw new Error(`env var ${name}=${v} is not a number`);
  return n;
}

export const isLocal = () =>
  (process.env.NODE_ENV ?? "").toLowerCase() === "local" ||
  process.env.ANNEMUSIC_LOCAL === "1";

File: packages/shared-py/shared/env.py
"""Env var access with explicit failure on missing required values."""

from __future__ import annotations

import os


def required(name: str) -> str:
    value = os.environ.get(name)
    if not value:
        raise RuntimeError(f"missing required env var: {name}")
    return value


def optional(name: str, fallback: str = "") -> str:
    return os.environ.get(name) or fallback


def optional_int(name: str, fallback: int) -> int:
    raw = os.environ.get(name)
    if raw is None or raw == "":
        return fallback
    try:
        return int(raw)
    except ValueError as exc:
        raise RuntimeError(f"env var {name}={raw!r} is not an int") from exc


def is_local() -> bool:
    return os.environ.get("NODE_ENV", "").lower() == "local" or os.environ.get("ANNEMUSIC_LOCAL") == "1"
